tests/src/python/test_qgsrulebasedrenderer.py

```diff
@@ -370,6 +377,87 @@ def testWillRenderFeatureTwoElse(self):
         self.assertEqual(rootrule.rulesForFeature(ft, ctx)[0], self.rx3)
         renderer.stopRender(ctx)
 
+    def testUsedAttributes(self):
+        ctx = QgsRenderContext.fromMapSettings(self.mapsettings)
+
+        # Create rulebased style
+        sym2 = QgsFillSymbol.createSimple({'color': '#71bd6c', 'outline_color': 'black'})
+        sym3 = QgsFillSymbol.createSimple({'color': '#1f78b4', 'outline_color': 'black'})
+
+        self.rx2 = QgsRuleBasedRenderer.Rule(sym2, 0, 0, '"id" = 200')
+        self.rx3 = QgsRuleBasedRenderer.Rule(sym3, 1000, 100000000, 'ELSE')
+
+        rootrule = QgsRuleBasedRenderer.Rule(None)
+        rootrule.appendChild(self.rx2)
+        rootrule.appendChild(self.rx3)
+
+        renderer = QgsRuleBasedRenderer(rootrule)
+
+        self.assertCountEqual(renderer.usedAttributes(ctx), {'id'})
+        self.assertIn('id', renderer.usedAttributes(ctx))
```

Comment:
This test is redundant -- it's covered by the assertCountEqual check (which despite the misleading name, actually checks if the contents of a list are equal while ignoring the order!).

Reply:
I did not know it. I was thinking that asserCountEqual is only about count and that it is better to have the same list to explain what the test are waiting for.
Thanks to pointing it. I will update the tests.

Reply:
I thought the same initially too. It's a really badly chosen name.

Reply:
Done, thanks

Reply:
Done, thanks